### PR TITLE
{AppConfig} Fix #13498958: Fix errors with az appconfig revision list when using customized fields

### PR DESCRIPTION
**Pull request by pratiksanglikar**
**Description**
Fixes [#13498958](https://msazure.visualstudio.com/Azure%20AppConfig/_workitems/edit/13498958) Fix errors with 'az appconfig revision list' when using customized fields

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py [hunk 1]
```diff
@@ -762,10 +762,15 @@ def list_revision(cmd,
...
+        query_fields = []
+        if fields:
+            for field in fields:
+                query_fields.append(field.name.lower())
+
         revisions_iterable = azconfig_client.list_revisions(key_filter=key,
                                                             label_filter=label,
                                                             accept_datetime=datetime,
-                                                            fields=fields)
+                                                            fields=query_fields)
```
Comment (shenmuxiaosen) [inline src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py:773]: >query_fields [](http://example.com/codeflow?start=67&length=12)

Based on the [sdk comments](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py#L454), we should leave none to include all fields.

File: src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py [hunk 2]
```diff
@@ -518,6 +518,75 @@ def test_resolve_keyvault(self, key_vault, resource_group):
...
+        })
+
+        self.cmd(
+            'appconfig kv set -n {config_store_name} --key {key} --value {value} --content-type {content_type} --label {label} -y',
+            checks=[self.check('contentType', entry_content_type),
+                    self.check('key', entry_key),
+                    self.check('value', updated_entry_value),
+                    self.check('label', updated_label)])
+
+        revisions = self.cmd('appconfig revision list -n {config_store_name} --key {key} --label * --top 2 --fields content_type etag key label last_modified locked tags value').get_output_in_json()
```
Comment (shenmuxiaosen) [inline src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py:580]: >content_type etag key label last_modified locked tags value [](http://example.com/codeflow?start=116&length=59)

nit: if you list all fields, this argument is mostly doing nothing. Can you just list a few then we can also check negative cases.

File: src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py (hunk 1 above)
Comment (pratiksanglikar) [reply src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py:774]: Thanks for pointing that out. Updated.

File: src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py [hunk 3]
```diff
@@ -518,6 +518,75 @@ def test_resolve_keyvault(self, key_vault, resource_group):
...
+
+        revisions = self.cmd('appconfig revision list -n {config_store_name} --key {key} --label * --top 2 --fields content_type etag label last_modified value').get_output_in_json()
+        assert len(revisions) == 2
+
+        assert revisions[0]['content_type'] == 'text'
+        assert revisions[0]['label'] == 'newlabel'
+        assert revisions[1]['label'] == 'v1.0.0'
+        assert revisions[0]['value'] == 'Green'
+        assert revisions[0]['last_modified'] is not None
+        assert revisions[1]['etag'] is not None
```
Comment (shenmuxiaosen) [inline src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py:588]: This doesn't look like a complete check. Check revision[0] content_type, etag, label, last_modified, value to be some value and no key. Then repeat for revision[1] if you want.

File: src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py (hunk 2 above)
Comment (pratiksanglikar) [reply src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py]: Good point. Updated.

File: src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py (hunk 3 above)
Comment (pratiksanglikar) [reply src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py:591]: I have updated the assertions to include all the values (present or missing)

**Review by shenmuxiaosen — APPROVED** [review]
:shipit:

**Comment by yonzhan** [comment]
AppConfig

**Review by wangzelin007 — APPROVED** [review]

**Comment by wangzelin007** [comment]
@pratiksanglikar,
Please re-record the failed tests.
image
And fix the conflicting file:
src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/export_features_prop.json

**Review by zhoxing-ms — APPROVED** [review]
@pratiksanglikar Could you solve the CI issue and conflicts by tomorrow? 
If not, this PR will not be able to catch up with the release of this sprint and can only be postponed to the next sprint (04-05)

**Comment by pratiksanglikar** [comment]
@zhoxing-ms let's merge this PR and take it in the current sprint.

**Pull request merged**